src/ansys/mapdl/core/krylov.py

```diff
@@ -0,0 +1,586 @@
+            f.close()
+
+        # Output generated subspace vectors to file
+        if out_key:
+            self._mapdl.run(f"*PRINT, {self.Qz.id}, {self.jobname}_Qz.txt")
+
+        return self.Qz
+
+    def krysolve(self, freq_start, freq_end, num_freq, load_key, out_key=False):
+        """Reduces system of equations and solve at each frequency
+
+        This method uses a KRYLOV subspace to solve a reduced harmonic
+        analysis over a specified frequency range for a given number of
+        frequency points (intervals).
+
+        Parameters
+        ----------
+        freq_start : int
+                    Starting value of the frequency range (Hz)
+        freq_end   : int
+                    Ending value of the frequency range (Hz)
+        num_freq   : int
+                    User specified number of intervals in frequency range
+        load_key   : int
+                    Key specifying whether load should be ramped(0) or stepped(1)
+        out_key    : Bool, optional
+                    Key to output reduced solution to Yz.txt file [Default:False]
+
+        Returns
+        -------
+        AnsMat
+            Reduced solution over Frequency range.
+
+        Notes
+        -----
+        Distributed ANSYS Restriction: This command is not supported in
+        Distributed ANSYS."""
+
+        # Store input arguments from user
+        self.freq_start = freq_start
+        self.num_freq = num_freq
+        self.load_key = load_key
+
+        # Check for illegal input values by the user
+        if not isinstance(self.freq_start, int) or self.freq_start < 0:
+            raise ValueError(
+                "The beginning frequency value for solving the reduced solution is required to be greater than or equal to 0"
+            )
+        if not isinstance(self.num_freq, int) or self.num_freq <= 0:
+            raise ValueError(
+                "The number of frequencies for which to compute the reduced solution is required to be greater than 0"
+            )
+        if not isinstance(self.load_key, int) or self.load_key < 0 or self.load_key > 1:
+            raise ValueError(
+                "The Load key value for computing the reduced solution is required to be 0 or 1"
+            )
+        if not isinstance(out_key, bool):
+            raise ValueError(
+                "The out_key value for computing the reduced solution is required to be Boolean True or False"
+            )
+
+        # Execute macro if no errors above
+        Az = self.mm.mat(name="Az")
+        nDOF = self._ndof
+        dim_q = self.dim_q
+
+        # Define solution variables
+        QtF = self.mm.zeros(dim_q, dtype=np.cdouble)
+        QtAQ = self.mm.zeros(dim_q, dim_q, dtype=np.cdouble)  # [QtAQ], reduced matrix
+        self.Yz = self.mm.zeros(
+            dim_q, self.num_freq, dtype=np.cdouble
+        )  # [Yz], reduced solution over range
+        QtA = self.mm.zeros(nDOF, dim_q, dtype=np.cdouble)  # [QtA]
```

Comment:
```suggestion
```
It seems to me ``QtA`` is not used anywhere. Do we need to calculate it?

Reply:
Yes you are right , I will remove QtA , looks like it has not been used anywhere.